pkg/tfpfbridge/schemashim/schema_decorators.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemashim
+
+import (
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+type schemaDecorator struct {
```

Comment:
Nit - thoughts on embedding the shim.Schema interface anonymously - in which case you wouldn't need to implement all the other methods?
```
type schemaWithOptional struct {
    shim.Schema // embedded interface
    optional func(shim.Schema) bool
}

func (o *schemaWithOptional) Optional() bool {
	if o.optional != nil {
		return o.optional(o.Schema)
	}
	return o.Schema.Optional()
}
```